modules/core/src/v2/coinFactory.ts

```diff
@@ -68,8 +69,15 @@ import { Bcha } from './coins/bcha';
 import { Tbcha } from './coins/tbcha';
 import { Dot } from './coins/dot';
 import { Tdot } from './coins/tdot';
+import { Environments } from './environments';
 
 export type CoinConstructor = (bitgo: BitGo, staticsCoin?: Readonly<StaticsBaseCoin>) => BaseCoin;
+const coinConstructorDual =
+  (coinConstructorMainNet: CoinConstructor, coinConstructorTestNet: CoinConstructor) =>
+  (bitgo: BitGo, staticsCoin?: Readonly<StaticsBaseCoin>) =>
+    Environments[bitgo.getEnv()].network !== 'testnet'
+      ? coinConstructorMainNet(bitgo, staticsCoin)
+      : coinConstructorTestNet(bitgo, staticsCoin);
```

Comment:
@shiksb @CapnMigraine We find this solution. But I don't like it. 
Could we talk about it?